[Search] Add ESQL syntax to semantic and hybrid search tutorials (#821)
elastic/docs-content: solutions/search/hybrid-semantic-text.md

## Perform hybrid search [hybrid-search-perform-search]

After reindexing the data into the `semantic-embeddings` index, you can perform hybrid search to combine semantic and lexical search results. Choose between [retrievers](retrievers-overview.md) or [{{esql}}](/explore-analyze/query-filter/languages/esql.md) syntax to execute the query.

::::{tab-set}
:group: query-type

> :::{tab-item} Retrievers

Author: 🤔 I'm not sure...would we be happy to put retrievers under the query DSL umbrella for simplicity here?

> :::{tab-item} Retrievers

Author: ah OK gotcha I thought we talked about retrievers as alternative `_search` syntax but not necessarily Query DSL 👍

> :::{tab-item} Retrievers

Author: https://github.com/elastic/docs-content/pull/821/commits/134163246239227c991d3afe7b29c6b46bc7e256

> :::{tab-item} Retrievers

Reviewer: Sorry for the confusion and thanks!

:sync: retrievers

> This example uses retrievers with [reciprocal rank fusion (RRF)](elasticsearch://reference/elasticsearch/rest-apis/reciprocal-rank-fusion.md). RRF is a technique that merges the rankings from both semantic and lexical queries, giving more weight to results that rank high in either search. This ensures that the final results are balanced and relevant.

Reviewer: Maybe add a link to retrievers as well

Author: Good call, I had one on line 105 but agree better to have it here 👍

4. The `semantic_text` field is used to perform the semantic search.

> After performing the hybrid search, the query will return the top 10 documents that match both semantic and lexical search criteria. The results include detailed information about each document.

Reviewer: It's not strictly necessary that results match both:

```suggestion
After performing the hybrid search, the query will return the combined top 10 documents for both semantic and lexical search criteria. The results include detailed information about each document.
```

Author: 👍
